test/test_post_processor.cpp

```diff
@@ -73,30 +83,122 @@ BOOST_AUTO_TEST_CASE(CoordinatesPassedThroughWhenNotFiltered) {
 
-  int body_part_index = BodyPartMin;
-  for (auto body_part : output.body_parts) {
-    BOOST_TEST(
-        check_coords_match(body_part, dummy_input.body_parts[body_part_index]));
-    body_part_index++;
-  }
+  check_coords_match(output.body_parts[Head], dummy_input.body_parts[head_top]);
+  check_coords_match(output.body_parts[Neck],
+                     dummy_input.body_parts[upper_neck]);
+  check_coords_match(output.body_parts[Shoulder],
+                     dummy_input.body_parts[right_shoulder]);
+  check_coords_match(output.body_parts[Hip], dummy_input.body_parts[right_hip]);
+  check_coords_match(output.body_parts[Knee],
+                     dummy_input.body_parts[right_knee]);
+  check_coords_match(output.body_parts[Foot],
+                     dummy_input.body_parts[right_ankle]);
+}
+
+BOOST_AUTO_TEST_CASE(AverageCoordinatesWhenBothTrustworthy) {
+  IIR::SmoothingSettings settings =
+      IIR::SmoothingSettings{std::vector<std::vector<float>>{}};
+  PostProcessing::PostProcessor post_proc =
+      PostProcessing::PostProcessor(0.0, settings);
+
+  Inference::InferenceResults dummy_input;
+  dummy_input.body_parts = {
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},  // Right shoulder
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{1, 1, 1.0},  // Left shoulder
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},  // Pelvis
+      Inference::Coordinate{0, 0, 1.0},  // Right hip
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0.5, 0.5, 1.0},  // Left hip
+      Inference::Coordinate{0, 0, 1.0},
+      Inference::Coordinate{0, 0, 1.0},
+  };
+
+  PostProcessing::ProcessedResults output = post_proc.run(dummy_input);
+
+  // Check shoulder mean works
+  float x = mean_of_coords_x(dummy_input.body_parts[left_shoulder],
+                             dummy_input.body_parts[right_shoulder]);
+  float y = mean_of_coords_y(dummy_input.body_parts[left_shoulder],
+                             dummy_input.body_parts[right_shoulder]);
+  BOOST_CHECK_CLOSE(x, output.body_parts[Shoulder].x, 0.1);
+  BOOST_CHECK_CLOSE(y, output.body_parts[Shoulder].y, 0.1);
+
+  // Check hip mean works
+  x = mean_of_coords_x(dummy_input.body_parts[left_hip],
+                       dummy_input.body_parts[right_hip]);
+  x = (dummy_input.body_parts[pelvis].x + x) / 2;
+  y = mean_of_coords_y(dummy_input.body_parts[left_hip],
+                       dummy_input.body_parts[right_hip]);
+  y = (dummy_input.body_parts[pelvis].y + y) / 2;
+  BOOST_CHECK_CLOSE(x, output.body_parts[Hip].x, 0.1);
+  BOOST_CHECK_CLOSE(y, output.body_parts[Hip].y, 0.1);
+}
+
+BOOST_AUTO_TEST_CASE(DontAverageCoordinatesWhenNotBothTrustworthy) {
```

Comment:
Is there a test case for one only one is trustworthy? Can't see it in the diff file here

Reply:
That's what this is supposed to be